### PR TITLE
chore(releasing): Prepare v0.24.0 release

### PR DESCRIPTION
**Pull request by jszwedko**
TODOs:

- [x] Review PRs since the v0.24 branch was cut to identify any that should be picked over
- [x] Cherry-pick [milestone issues](https://github.com/vectordotdev/vector/milestone/93)
  - [x] Pending https://github.com/vectordotdev/vector/issues/13461
- [x] Add What's Next
- [x] Upgrade note about removal of deprecated transforms
- [x] Deprecation of `geoip` transform
- [x] ~~Release highlight for OpenTelemetry source?~~
- [x] Fix docs publishing for GCP Chronicle sink
- [x] Should we make metrics expiration default?
- [x] Flesh out description

Reviewers should check:

- Milestone issues were picked (https://github.com/vectordotdev/vector/milestone/93)
- Read over the deployed release notes
- Review the `Prepare v0.24.0` release commit

### CODE REVIEW

File: website/cue/reference/releases/0.24.0.cue [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+		breaking changes in this release.
+
+		In addition to the new features, enhancements, and fixes listed below, this release adds:
+
+		- A new `axiom` sink for sending events to Axiom
+		- A new `gcp_chronicle_unstructured` sink for sending unstructured log events to GCP Chronicle
+		- A new `file_descriptor` source to consume input from file descriptors
+		- A new `opentelemetry` source to receive input from OpenTelemetry collectors and SDKs. Only
+		  logs are supported in this release, but support for metrics and traces are in-flight.
+		  A `opentelemetry` sink will follow.
```
Comment (neuronull) [inline website/cue/reference/releases/0.24.0.cue:35]:
```suggestion
		  An `opentelemetry` sink will follow.
```

File: website/cue/reference/releases/0.24.0.cue [hunk 2]
```diff
@@ -0,0 +1,715 @@
...
+
+		Note that this release as has a backwards incompatible data model change that users of the
```
Comment (neuronull) [inline website/cue/reference/releases/0.24.0.cue:37]:
```suggestion
		Note that this release has a backwards incompatible data model change that users of the
```

File: website/cue/reference/releases/0.24.0.cue [hunk 3]
```diff
@@ -0,0 +1,715 @@
...
+		upgrade guide](/highlights/2022-08-16-0-24-0-upgrade-guide#metric-buckets) for more
+		details.
+		"""
+
+	changelog: [
+		{
+			type: "enhancement"
+			scopes: ["vrl", "vrl: compiler"]
+			description: """
+				The VRL compiler now rejects assignments to fields on values known not to be an
```
Comment (neuronull) [inline website/cue/reference/releases/0.24.0.cue:48]:
```suggestion
				The VRL compiler now rejects assignments to fields on values known not to be
```

File: website/cue/reference/releases/0.24.0.cue [hunk 4]
```diff
@@ -0,0 +1,715 @@
...
+				buckets, allowing it to avoid errors when scraping endpoints that have buckets with
+				very counts that didn't fit in an unsigned 32-bit integer.
+				"""
+			pr_numbers: [13318]
+		},
+		{
+			type: "enhancement"
+			scopes: ["vrl", "vrl: stdlib"]
+			description: """
+				A `filter` function was added to VRL to allow easily remove keys from objects or
```
Comment (neuronull) [inline website/cue/reference/releases/0.24.0.cue:91]:
```suggestion
				A `filter` function was added to VRL to allow easy removal of keys from objects or
```

File: website/cue/reference/releases/0.24.0.cue [hunk 5]
```diff
@@ -0,0 +1,715 @@
...
+			contributors: ["bahlo"]
+			pr_numbers: [13007]
+		},
+		{
+			type: "feat"
+			scopes: ["codecs"]
+			description: """
+				A new `gelf` codec was added for decoding/encoding
+				[GELF](https://docs.graylog.org/docs/gelf) data in Vector's sources and sinks. It
+				can be used used via `encoding.codec` on sinks and `decoding.codec` on sources, for
```
Comment (neuronull) [inline website/cue/reference/releases/0.24.0.cue:117]:
```suggestion
				can be used via `encoding.codec` on sinks and `decoding.codec` on sources, for
```

File: website/cue/reference/releases/0.24.0.cue [hunk 6]
```diff
@@ -0,0 +1,715 @@
...
+				"""
+			contributors: ["mcasper"]
+			pr_numbers: [13436]
+		},
+		{
+			type: "enhancement"
+			scopes: ["aws provider"]
+			description: """
+				AWS components now allow specifying a different region to use when assuming role
+				than the region otherwise configured on the component via `auth.region`.
```
Comment (neuronull) [inline website/cue/reference/releases/0.24.0.cue:323]: The bit "when assuming role than" reads weird for me but maybe that is just me not knowing much about AWS.

File: website/cue/reference/releases/0.24.0.cue [hunk 7]
```diff
@@ -0,0 +1,715 @@
...
+				a histogram of the time difference of when Vector ingests an event and the timestamp
+				of the event itself (if it exists).
+				"""
+			pr_numbers: [13611]
+		},
+		{
+			type: "enhancement"
+			scopes: ["internal_metrics source"]
+			description: """
+				The `internal_metrics` source now emits a `internal_metrics_cardinality` gauge
```
Comment (neuronull) [inline website/cue/reference/releases/0.24.0.cue:361]:
```suggestion
				The `internal_metrics` source now emits an `internal_metrics_cardinality` gauge
```

**Comment by neuronull** [comment]
> Milestone issues were picked (https://github.com/vectordotdev/vector/milestone/93)

Looks correct to me (minus the pending one)

File: website/cue/reference/releases/0.24.0.cue (hunk 6 above)
Comment (jszwedko) [reply website/cue/reference/releases/0.24.0.cue:323]: Yeah, it is a little awkward. I'll reword.

**Comment by jszwedko** [comment]
Thanks for the thorough review @neuronull !

**Review by neuronull — APPROVED** [review]
LGMT :rocket:

File: website/content/en/highlights/2022-08-16-0-24-0-upgrade-guide.md [hunk 8]
```diff
@@ -167,3 +196,48 @@ metric described above is now deprecated and will be removed in a future version
...
 for additional details.
+
+
+#### Deprecation of modulus operator {#mod-deprecation}
+
+The modulus operator (`%`) is being deprecated in `0.24.0` in favor of the newly introduced `mod`
+function. This operator is fairly infrequently used, and moving this to a function allows the `%`
+to be repurposed for an upcoming feature, allowing you to query event metadata similar to event fields. This also now additionally allows the option of simply aborting
+the `mod` function on an error instead of either capturing the error or using the "error coalescing"
+operator. The modulus operator will be completely removed as early as `0.25.0`.
```
Comment (spencergilbert) [inline website/content/en/highlights/2022-08-16-0-24-0-upgrade-guide.md:207]: Probably too long of a line

**Review by spencergilbert — APPROVED** [review]

**Comment by jszwedko** [comment]
Merged manually to preserve commit history.

**Pull request closed without merging**